File: source/AdventureWorksAIWorkspaceAPI/src/Application/Common/Dtos/Charts/ChartKind.cs
namespace AdventureWorksAIWorkspaceAPI.Application.Common.Dtos.Charts;

/// <summary>
/// The kind of visualization suggested for a report result. Values map to MUI X Charts component
/// types on the frontend.
/// </summary>
public enum ChartKind
{
    /// <summary>No meaningful chart; render the rows as a table.</summary>
    Table,

    /// <summary>Category comparison (MUI BarChart).</summary>
    Bar,

    /// <summary>Trend over an ordered/date axis (MUI LineChart).</summary>
    Line,

    /// <summary>Parts of a whole (MUI PieChart).</summary>
    Pie,

    /// <summary>Filled trend over an ordered axis (MUI LineChart with area).</summary>
    Area
}

File: source/web/src/features/reports/lib/chart-kind.ts
import type { ChartSpec } from './report-types'

const chartKindLabels: Record<string, string> = {
  '0': 'table',
  '1': 'bar',
  '2': 'line',
  '3': 'pie',
  '4': 'area',
}

export function getChartKind(spec: Pick<ChartSpec, 'kind'>): string {
  return chartKindLabels[String(spec.kind)] ?? String(spec.kind).toLowerCase()
}

export function isTableChart(spec: Pick<ChartSpec, 'kind'>): boolean {
  return getChartKind(spec) === 'table'
}
